# Imports

In [1]:
import pandas as pd
import yfinance as yf
import warnings
warnings.filterwarnings("ignore")

# Load Membership Panel

In [2]:
panel = pd.read_csv(r"..\..\Data\Main\membership_panel_clean.csv")
print(f"Tickers to download: {len(panel)}")
print(panel.head())

Tickers to download: 606
  yf_ticker                         name  first_seen   last_seen  \
0         A     Agilent Technologies Inc  2020-01-02  2025-12-31   
1       AAL  American Airlines Group Inc  2020-01-02  2024-07-01   
2       AAP       Advance Auto Parts Inc  2020-01-02  2023-07-03   
3      AAPL                    Apple Inc  2020-01-02  2025-12-31   
4      ABBV                   AbbVie Inc  2020-01-02  2025-12-31   

   quarters_present  
0                24  
1                19  
2                15  
3                24  
4                24  


# Download Daily Prices from yfinance

Each ticker is downloaded from its first_seen date to last_seen + 1 day (to ensure full coverage at panel edges). Tickers that fail are typically acquired or delisted companies whose data has been removed from Yahoo Finance.

In [5]:
import csv

tickers = panel['yf_ticker'].tolist()
success = []
failed = []
first_ticker = True

output_path = r"..\..\Data\Main\sp500_prices_2020_2025.csv"

for i, t in enumerate(tickers):
    row = panel[panel['yf_ticker'] == t].iloc[0]
    start = pd.to_datetime(row['first_seen'])
    end = pd.to_datetime(row['last_seen']) + pd.DateOffset(days=1)

    print(f"{i+1}/{len(tickers)} — {t} ({start.date()} to {end.date()})", end=" ... ")
    try:
        data = yf.download(t, start=start, end=end, progress=False)
        close = data['Close'].dropna()
        if len(close) == 0:
            failed.append(t)
            print("FAILED")
        else:
            first = close.index[0].strftime('%Y-%m-%d')
            last = close.index[-1].strftime('%Y-%m-%d')
            success.append({'ticker': t, 'first_date': first, 'last_date': last, 'days': len(close)})
            
            # Flatten columns and add ticker
            data = data.reset_index()
            data.columns = [c[0] if isinstance(c, tuple) else c for c in data.columns]
            data.rename(columns={'Date': 'date', 'Price': 'date'}, inplace=True)
            data['ticker'] = t
            
            # Write header only for first ticker
            data.to_csv(output_path, mode='a' if not first_ticker else 'w', 
                       header=first_ticker, index=False)
            first_ticker = False
            
            print(f"OK ({len(close)} days)")
    except:
        failed.append(t)
        print("FAILED")

print(f"\nSuccess: {len(success)}")
print(f"Failed: {len(failed)}")
print(f"Failed tickers: {failed}")

success_df = pd.DataFrame(success)

1/606 — A (2020-01-02 to 2026-01-01) ... OK (1508 days)
2/606 — AAL (2020-01-02 to 2024-07-02) ... OK (1131 days)
3/606 — AAP (2020-01-02 to 2023-07-04) ... OK (881 days)
4/606 — AAPL (2020-01-02 to 2026-01-01) ... OK (1508 days)
5/606 — ABBV (2020-01-02 to 2026-01-01) ... OK (1508 days)
6/606 — ABMD (2020-01-02 to 2022-10-04) ... 

$ABMD: possibly delisted; no timezone found

1 Failed download:
['ABMD']: possibly delisted; no timezone found


FAILED
7/606 — ABNB (2023-10-02 to 2026-01-01) ... OK (565 days)
8/606 — ABT (2020-01-02 to 2026-01-01) ... OK (1508 days)
9/606 — ACGL (2023-01-03 to 2026-01-01) ... OK (752 days)
10/606 — ACN (2020-01-02 to 2026-01-01) ... OK (1508 days)
11/606 — ADBE (2020-01-02 to 2026-01-01) ... OK (1508 days)
12/606 — ADI (2020-01-02 to 2026-01-01) ... OK (1508 days)
13/606 — ADM (2020-01-02 to 2026-01-01) ... OK (1508 days)
14/606 — ADP (2020-01-02 to 2026-01-01) ... OK (1508 days)
15/606 — ADSK (2020-01-02 to 2026-01-01) ... OK (1508 days)
16/606 — AEE (2020-01-02 to 2026-01-01) ... OK (1508 days)
17/606 — AEP (2020-01-02 to 2026-01-01) ... OK (1508 days)
18/606 — AES (2020-01-02 to 2026-01-01) ... OK (1508 days)
19/606 — AFL (2020-01-02 to 2026-01-01) ... OK (1508 days)
20/606 — AGN (2020-01-02 to 2020-04-02) ... 

$AGN: possibly delisted; no timezone found

1 Failed download:
['AGN']: possibly delisted; no timezone found


FAILED
21/606 — AIG (2020-01-02 to 2026-01-01) ... OK (1508 days)
22/606 — AIV (2020-01-02 to 2020-10-02) ... OK (190 days)
23/606 — AIZ (2020-01-02 to 2026-01-01) ... OK (1508 days)
24/606 — AJG (2020-01-02 to 2026-01-01) ... OK (1508 days)
25/606 — AKAM (2020-01-02 to 2026-01-01) ... OK (1508 days)
26/606 — ALB (2020-01-02 to 2026-01-01) ... OK (1508 days)
27/606 — ALGN (2020-01-02 to 2026-01-01) ... OK (1508 days)
28/606 — ALK (2020-01-02 to 2023-10-03) ... OK (944 days)
29/606 — ALL (2020-01-02 to 2026-01-01) ... OK (1508 days)
30/606 — ALLE (2020-01-02 to 2026-01-01) ... OK (1508 days)
31/606 — ALXN (2020-01-02 to 2021-07-02) ... 

$ALXN: possibly delisted; no timezone found

1 Failed download:
['ALXN']: possibly delisted; no timezone found


FAILED
32/606 — AMAT (2020-01-02 to 2026-01-01) ... OK (1508 days)
33/606 — AMCR (2020-01-02 to 2026-01-01) ... OK (1508 days)
34/606 — AMD (2020-01-02 to 2026-01-01) ... OK (1508 days)
35/606 — AME (2020-01-02 to 2026-01-01) ... OK (1508 days)
36/606 — AMGN (2020-01-02 to 2026-01-01) ... OK (1508 days)
37/606 — AMP (2020-01-02 to 2026-01-01) ... OK (1508 days)
38/606 — AMT (2020-01-02 to 2026-01-01) ... OK (1508 days)
39/606 — AMTM (2024-10-01 to 2024-10-02) ... OK (1 days)
40/606 — AMZN (2020-01-02 to 2026-01-01) ... OK (1508 days)
41/606 — ANET (2020-01-02 to 2026-01-01) ... OK (1508 days)
42/606 — ANSS (2020-01-02 to 2025-07-02) ... 

$ANSS: possibly delisted; no timezone found

1 Failed download:
['ANSS']: possibly delisted; no timezone found


FAILED
43/606 — AON (2020-01-02 to 2026-01-01) ... OK (1508 days)
44/606 — AOS (2020-01-02 to 2026-01-01) ... OK (1508 days)
45/606 — APA (2020-01-02 to 2026-01-01) ... OK (1508 days)
46/606 — APD (2020-01-02 to 2026-01-01) ... OK (1508 days)
47/606 — APH (2020-01-02 to 2026-01-01) ... OK (1508 days)
48/606 — APO (2025-01-02 to 2026-01-01) ... OK (250 days)
49/606 — APP (2025-10-01 to 2026-01-01) ... OK (64 days)
50/606 — APTV (2020-01-02 to 2026-01-01) ... OK (1508 days)
51/606 — ARE (2020-01-02 to 2026-01-01) ... OK (1508 days)
52/606 — ARNC (2020-04-01 to 2020-04-02) ... 

$ARNC: possibly delisted; no timezone found

1 Failed download:
['ARNC']: possibly delisted; no timezone found


FAILED
53/606 — ATO (2020-01-02 to 2026-01-01) ... OK (1508 days)
54/606 — ATVI (2020-01-02 to 2023-10-03) ... 

$ATVI: possibly delisted; no timezone found

1 Failed download:
['ATVI']: possibly delisted; no timezone found


FAILED
55/606 — AVB (2020-01-02 to 2026-01-01) ... OK (1508 days)
56/606 — AVGO (2020-01-02 to 2026-01-01) ... OK (1508 days)
57/606 — AVY (2020-01-02 to 2026-01-01) ... OK (1508 days)
58/606 — AWK (2020-01-02 to 2026-01-01) ... OK (1508 days)
59/606 — AXON (2023-07-03 to 2026-01-01) ... OK (628 days)
60/606 — AXP (2020-01-02 to 2026-01-01) ... OK (1508 days)
61/606 — AZO (2020-01-02 to 2026-01-01) ... OK (1508 days)
62/606 — BA (2020-01-02 to 2026-01-01) ... OK (1508 days)
63/606 — BAC (2020-01-02 to 2026-01-01) ... OK (1508 days)
64/606 — BALL (2020-01-02 to 2026-01-01) ... OK (1508 days)
65/606 — BAX (2020-01-02 to 2026-01-01) ... OK (1508 days)
66/606 — BBWI (2020-01-02 to 2024-07-02) ... OK (1131 days)
67/606 — BBY (2020-01-02 to 2026-01-01) ... OK (1508 days)
68/606 — BDX (2020-01-02 to 2026-01-01) ... OK (1508 days)
69/606 — BEN (2020-01-02 to 2026-01-01) ... OK (1508 days)
70/606 — BF-B (2020-01-02 to 2026-01-01) ... OK (1508 days)
71/606 — BFH (2020-01-02 to 2020-04-02) ... OK

$CERN: possibly delisted; no timezone found

1 Failed download:
['CERN']: possibly delisted; no timezone found


FAILED
103/606 — CF (2020-01-02 to 2026-01-01) ... OK (1508 days)
104/606 — CFG (2020-01-02 to 2026-01-01) ... OK (1508 days)
105/606 — CHD (2020-01-02 to 2026-01-01) ... OK (1508 days)
106/606 — CHRW (2020-01-02 to 2026-01-01) ... OK (1508 days)
107/606 — CHTR (2020-01-02 to 2026-01-01) ... OK (1508 days)
108/606 — CI (2020-01-02 to 2026-01-01) ... OK (1508 days)
109/606 — CINF (2020-01-02 to 2026-01-01) ... OK (1508 days)
110/606 — CL (2020-01-02 to 2026-01-01) ... OK (1508 days)
111/606 — CLX (2020-01-02 to 2026-01-01) ... OK (1508 days)
112/606 — CMA (2020-01-02 to 2024-04-02) ... 

$CMA: possibly delisted; no price data found  (1d 2020-01-02 00:00:00 -> 2024-04-02 00:00:00) (Yahoo error = "No data found, symbol may be delisted")

1 Failed download:
['CMA']: possibly delisted; no price data found  (1d 2020-01-02 00:00:00 -> 2024-04-02 00:00:00) (Yahoo error = "No data found, symbol may be delisted")


FAILED
113/606 — CMCSA (2020-01-02 to 2026-01-01) ... OK (1508 days)
114/606 — CME (2020-01-02 to 2026-01-01) ... OK (1508 days)
115/606 — CMG (2020-01-02 to 2026-01-01) ... OK (1508 days)
116/606 — CMI (2020-01-02 to 2026-01-01) ... OK (1508 days)
117/606 — CMS (2020-01-02 to 2026-01-01) ... OK (1508 days)
118/606 — CNC (2020-01-02 to 2026-01-01) ... OK (1508 days)
119/606 — CNP (2020-01-02 to 2026-01-01) ... OK (1508 days)
120/606 — COF (2020-01-02 to 2026-01-01) ... OK (1508 days)
121/606 — COIN (2025-07-01 to 2026-01-01) ... OK (128 days)
122/606 — COO (2020-01-02 to 2026-01-01) ... OK (1508 days)
123/606 — COP (2020-01-02 to 2026-01-01) ... OK (1508 days)
124/606 — COR (2020-01-02 to 2026-01-01) ... OK (1508 days)
125/606 — COST (2020-01-02 to 2026-01-01) ... OK (1508 days)
126/606 — COTY (2020-01-02 to 2020-07-02) ... OK (126 days)
127/606 — CPAY (2020-01-02 to 2026-01-01) ... OK (1508 days)
128/606 — CPB (2020-01-02 to 2026-01-01) ... OK (1508 days)
129/606 — CPRI (2020-01-02 to

$CTLT: possibly delisted; no timezone found

1 Failed download:
['CTLT']: possibly delisted; no timezone found


FAILED
140/606 — CTRA (2020-01-02 to 2026-01-01) ... OK (1508 days)
141/606 — CTSH (2020-01-02 to 2026-01-01) ... OK (1508 days)
142/606 — CTVA (2020-01-02 to 2026-01-01) ... OK (1508 days)
143/606 — CTXS (2020-01-02 to 2022-07-02) ... 

$CTXS: possibly delisted; no timezone found

1 Failed download:
['CTXS']: possibly delisted; no timezone found


FAILED
144/606 — CVS (2020-01-02 to 2026-01-01) ... OK (1508 days)
145/606 — CVX (2020-01-02 to 2026-01-01) ... OK (1508 days)
146/606 — CXO (2020-01-02 to 2021-01-05) ... 

$CXO: possibly delisted; no timezone found

1 Failed download:
['CXO']: possibly delisted; no timezone found


FAILED
147/606 — CZR (2021-04-01 to 2025-07-02) ... OK (1067 days)
148/606 — D (2020-01-02 to 2026-01-01) ... OK (1508 days)
149/606 — DAL (2020-01-02 to 2026-01-01) ... OK (1508 days)
150/606 — DASH (2025-04-01 to 2026-01-01) ... OK (190 days)
151/606 — DAY (2021-10-01 to 2026-01-01) ... 

$DAY: possibly delisted; no price data found  (1d 2021-10-01 00:00:00 -> 2026-01-01 00:00:00) (Yahoo error = "No data found, symbol may be delisted")

1 Failed download:
['DAY']: possibly delisted; no price data found  (1d 2021-10-01 00:00:00 -> 2026-01-01 00:00:00) (Yahoo error = "No data found, symbol may be delisted")


FAILED
152/606 — DD (2020-01-02 to 2026-01-01) ... OK (1508 days)
153/606 — DDOG (2025-10-01 to 2026-01-01) ... OK (64 days)
154/606 — DE (2020-01-02 to 2026-01-01) ... OK (1508 days)
155/606 — DECK (2024-04-01 to 2026-01-01) ... OK (441 days)
156/606 — DELL (2024-10-01 to 2026-01-01) ... OK (314 days)
157/606 — DFS (2020-01-02 to 2025-04-02) ... 

$DFS: possibly delisted; no timezone found

1 Failed download:
['DFS']: possibly delisted; no timezone found


FAILED
158/606 — DG (2020-01-02 to 2026-01-01) ... OK (1508 days)
159/606 — DGX (2020-01-02 to 2026-01-01) ... OK (1508 days)
160/606 — DHI (2020-01-02 to 2026-01-01) ... OK (1508 days)
161/606 — DHR (2020-01-02 to 2026-01-01) ... OK (1508 days)
162/606 — DIS (2020-01-02 to 2026-01-01) ... OK (1508 days)
163/606 — DISCA (2020-01-02 to 2022-04-02) ... 

$DISCA: possibly delisted; no timezone found

1 Failed download:
['DISCA']: possibly delisted; no timezone found


FAILED
164/606 — DISCK (2020-01-02 to 2022-04-02) ... 

$DISCK: possibly delisted; no timezone found

1 Failed download:
['DISCK']: possibly delisted; no timezone found


FAILED
165/606 — DISH (2020-01-02 to 2023-04-04) ... 

$DISH: possibly delisted; no timezone found

1 Failed download:
['DISH']: possibly delisted; no timezone found


FAILED
166/606 — DLR (2020-01-02 to 2026-01-01) ... OK (1508 days)
167/606 — DLTR (2020-01-02 to 2026-01-01) ... OK (1508 days)
168/606 — DOC (2020-01-02 to 2026-01-01) ... OK (1508 days)
169/606 — DOV (2020-01-02 to 2026-01-01) ... OK (1508 days)
170/606 — DOW (2020-01-02 to 2026-01-01) ... OK (1508 days)
171/606 — DPZ (2020-07-01 to 2026-01-01) ... OK (1383 days)
172/606 — DRE (2020-01-02 to 2022-07-02) ... 

$DRE: possibly delisted; no timezone found

1 Failed download:
['DRE']: possibly delisted; no timezone found


FAILED
173/606 — DRI (2020-01-02 to 2026-01-01) ... OK (1508 days)
174/606 — DTE (2020-01-02 to 2026-01-01) ... OK (1508 days)
175/606 — DTM (2021-07-01 to 2021-07-02) ... OK (1 days)
176/606 — DUK (2020-01-02 to 2026-01-01) ... OK (1508 days)
177/606 — DVA (2020-01-02 to 2026-01-01) ... OK (1508 days)
178/606 — DVN (2020-01-02 to 2026-01-01) ... OK (1508 days)
179/606 — DXC (2020-01-02 to 2023-10-03) ... OK (944 days)
180/606 — DXCM (2020-07-01 to 2026-01-01) ... OK (1383 days)
181/606 — EA (2020-01-02 to 2026-01-01) ... OK (1508 days)
182/606 — EBAY (2020-01-02 to 2026-01-01) ... OK (1508 days)
183/606 — ECL (2020-01-02 to 2026-01-01) ... OK (1508 days)
184/606 — ED (2020-01-02 to 2026-01-01) ... OK (1508 days)
185/606 — EFX (2020-01-02 to 2026-01-01) ... OK (1508 days)
186/606 — EG (2020-01-02 to 2026-01-01) ... OK (1508 days)
187/606 — EIX (2020-01-02 to 2026-01-01) ... OK (1508 days)
188/606 — EL (2020-01-02 to 2026-01-01) ... OK (1508 days)
189/606 — ELV (2020-01-02 to 2026-01-01

$ETFC: possibly delisted; no timezone found

1 Failed download:
['ETFC']: possibly delisted; no timezone found


FAILED
204/606 — ETN (2020-01-02 to 2026-01-01) ... OK (1508 days)
205/606 — ETR (2020-01-02 to 2026-01-01) ... OK (1508 days)
206/606 — ETSY (2020-10-01 to 2024-07-02) ... OK (942 days)
207/606 — EVRG (2020-01-02 to 2026-01-01) ... OK (1508 days)
208/606 — EW (2020-01-02 to 2026-01-01) ... OK (1508 days)
209/606 — EXC (2020-01-02 to 2026-01-01) ... OK (1508 days)
210/606 — EXE (2025-04-01 to 2026-01-01) ... OK (190 days)
211/606 — EXPD (2020-01-02 to 2026-01-01) ... OK (1508 days)
212/606 — EXPE (2020-01-02 to 2026-01-01) ... OK (1508 days)
213/606 — EXR (2020-01-02 to 2026-01-01) ... OK (1508 days)
214/606 — F (2020-01-02 to 2026-01-01) ... OK (1508 days)
215/606 — FANG (2020-01-02 to 2026-01-01) ... OK (1508 days)
216/606 — FAST (2020-01-02 to 2026-01-01) ... OK (1508 days)
217/606 — FBIN (2020-01-02 to 2022-10-04) ... OK (694 days)
218/606 — FCX (2020-01-02 to 2026-01-01) ... OK (1508 days)
219/606 — FDS (2022-01-03 to 2026-01-01) ... OK (1003 days)
220/606 — FDX (2020-01-02 to 202

$FLIR: possibly delisted; no timezone found

1 Failed download:
['FLIR']: possibly delisted; no timezone found


FAILED
228/606 — FLS (2020-01-02 to 2021-01-05) ... OK (254 days)
229/606 — FMC (2020-01-02 to 2025-01-03) ... OK (1259 days)
230/606 — FOX (2020-01-02 to 2026-01-01) ... OK (1508 days)
231/606 — FOXA (2020-01-02 to 2026-01-01) ... OK (1508 days)
232/606 — FRCB (2020-01-02 to 2023-04-04) ... OK (819 days)
233/606 — FRT (2020-01-02 to 2026-01-01) ... OK (1508 days)
234/606 — FSLR (2023-01-03 to 2026-01-01) ... OK (752 days)
235/606 — FTI (2020-01-02 to 2021-01-05) ... OK (254 days)
236/606 — FTNT (2020-01-02 to 2026-01-01) ... OK (1508 days)
237/606 — FTRE (2023-07-03 to 2023-07-04) ... OK (1 days)
238/606 — FTV (2020-01-02 to 2026-01-01) ... OK (1508 days)
239/606 — GAP (2020-01-02 to 2022-01-04) ... OK (506 days)
240/606 — GD (2020-01-02 to 2026-01-01) ... OK (1508 days)
241/606 — GDDY (2024-07-01 to 2026-01-01) ... OK (378 days)
242/606 — GE (2020-01-02 to 2026-01-01) ... OK (1508 days)
243/606 — GEHC (2023-04-03 to 2026-01-01) ... OK (690 days)
244/606 — GEN (2020-01-02 to 2026-01-0

$HBI: possibly delisted; no timezone found

1 Failed download:
['HBI']: possibly delisted; no timezone found


FAILED
263/606 — HCA (2020-01-02 to 2026-01-01) ... OK (1508 days)
264/606 — HD (2020-01-02 to 2026-01-01) ... OK (1508 days)
265/606 — HES (2020-01-02 to 2025-07-02) ... 

$HES: possibly delisted; no timezone found

1 Failed download:
['HES']: possibly delisted; no timezone found


FAILED
266/606 — HFC (2020-01-02 to 2021-04-02) ... 

$HFC: possibly delisted; no timezone found

1 Failed download:
['HFC']: possibly delisted; no timezone found


FAILED
267/606 — HIG (2020-01-02 to 2026-01-01) ... OK (1508 days)
268/606 — HII (2020-01-02 to 2026-01-01) ... OK (1508 days)
269/606 — HLT (2020-01-02 to 2026-01-01) ... OK (1508 days)
270/606 — HOG (2020-01-02 to 2020-04-02) ... OK (63 days)
271/606 — HOLX (2020-01-02 to 2026-01-01) ... OK (1508 days)
272/606 — HON (2020-01-02 to 2026-01-01) ... OK (1508 days)
273/606 — HOOD (2025-10-01 to 2026-01-01) ... OK (64 days)
274/606 — HP (2020-01-02 to 2020-04-02) ... OK (63 days)
275/606 — HPE (2020-01-02 to 2026-01-01) ... OK (1508 days)
276/606 — HPQ (2020-01-02 to 2026-01-01) ... OK (1508 days)
277/606 — HRB (2020-01-02 to 2020-07-02) ... OK (126 days)
278/606 — HRL (2020-01-02 to 2026-01-01) ... OK (1508 days)
279/606 — HSIC (2020-01-02 to 2026-01-01) ... OK (1508 days)
280/606 — HST (2020-01-02 to 2026-01-01) ... OK (1508 days)
281/606 — HSY (2020-01-02 to 2026-01-01) ... OK (1508 days)
282/606 — HUBB (2024-01-02 to 2026-01-01) ... OK (502 days)
283/606 — HUM (2020-01-02 to 2026-01-0

$IPG: possibly delisted; no timezone found

1 Failed download:
['IPG']: possibly delisted; no timezone found


FAILED
298/606 — IPGP (2020-01-02 to 2022-04-02) ... OK (568 days)
299/606 — IQV (2020-01-02 to 2026-01-01) ... OK (1508 days)
300/606 — IR (2020-04-01 to 2026-01-01) ... OK (1446 days)
301/606 — IRM (2020-01-02 to 2026-01-01) ... OK (1508 days)
302/606 — ISRG (2020-01-02 to 2026-01-01) ... OK (1508 days)
303/606 — IT (2020-01-02 to 2026-01-01) ... OK (1508 days)
304/606 — ITW (2020-01-02 to 2026-01-01) ... OK (1508 days)
305/606 — IVZ (2020-01-02 to 2026-01-01) ... OK (1508 days)
306/606 — J (2020-01-02 to 2026-01-01) ... OK (1508 days)
307/606 — JBHT (2020-01-02 to 2026-01-01) ... OK (1508 days)
308/606 — JBL (2024-01-02 to 2026-01-01) ... OK (502 days)
309/606 — JCI (2020-01-02 to 2026-01-01) ... OK (1508 days)
310/606 — JKHY (2020-01-02 to 2026-01-01) ... OK (1508 days)
311/606 — JNJ (2020-01-02 to 2026-01-01) ... OK (1508 days)
312/606 — JNPR (2020-01-02 to 2025-07-02) ... 

$JNPR: possibly delisted; no timezone found

1 Failed download:
['JNPR']: possibly delisted; no timezone found


FAILED
313/606 — JPM (2020-01-02 to 2026-01-01) ... OK (1508 days)
314/606 — JWN (2020-01-02 to 2020-04-02) ... 

$JWN: possibly delisted; no timezone found

1 Failed download:
['JWN']: possibly delisted; no timezone found


FAILED
315/606 — K (2020-01-02 to 2026-01-01) ... 

$K: possibly delisted; no timezone found

1 Failed download:
['K']: possibly delisted; no timezone found


FAILED
316/606 — KDP (2022-07-01 to 2026-01-01) ... OK (879 days)
317/606 — KEY (2020-01-02 to 2026-01-01) ... OK (1508 days)
318/606 — KEYS (2020-01-02 to 2026-01-01) ... OK (1508 days)
319/606 — KHC (2020-01-02 to 2026-01-01) ... OK (1508 days)
320/606 — KIM (2020-01-02 to 2026-01-01) ... OK (1508 days)
321/606 — KKR (2024-07-01 to 2026-01-01) ... OK (378 days)
322/606 — KLAC (2020-01-02 to 2026-01-01) ... OK (1508 days)
323/606 — KLG (2023-10-02 to 2023-10-03) ... 

$KLG: possibly delisted; no timezone found

1 Failed download:
['KLG']: possibly delisted; no timezone found


FAILED
324/606 — KMB (2020-01-02 to 2026-01-01) ... OK (1508 days)
325/606 — KMI (2020-01-02 to 2026-01-01) ... OK (1508 days)
326/606 — KMX (2020-01-02 to 2026-01-01) ... OK (1508 days)
327/606 — KO (2020-01-02 to 2026-01-01) ... OK (1508 days)
328/606 — KR (2020-01-02 to 2026-01-01) ... OK (1508 days)
329/606 — KSS (2020-01-02 to 2020-07-02) ... OK (126 days)
330/606 — KSU (2020-01-02 to 2021-10-02) ... 

$KSU: possibly delisted; no timezone found

1 Failed download:
['KSU']: possibly delisted; no timezone found


FAILED
331/606 — KVUE (2023-10-02 to 2026-01-01) ... OK (565 days)
332/606 — L (2020-01-02 to 2026-01-01) ... OK (1508 days)
333/606 — LDOS (2020-01-02 to 2026-01-01) ... OK (1508 days)
334/606 — LEG (2020-01-02 to 2021-10-02) ... OK (442 days)
335/606 — LEN (2020-01-02 to 2026-01-01) ... OK (1508 days)
336/606 — LH (2020-01-02 to 2026-01-01) ... OK (1508 days)
337/606 — LHX (2020-01-02 to 2026-01-01) ... OK (1508 days)
338/606 — LII (2025-01-02 to 2026-01-01) ... OK (250 days)
339/606 — LIN (2020-01-02 to 2026-01-01) ... OK (1508 days)
340/606 — LKQ (2020-01-02 to 2026-01-01) ... OK (1508 days)
341/606 — LLY (2020-01-02 to 2026-01-01) ... OK (1508 days)
342/606 — LMT (2020-01-02 to 2026-01-01) ... OK (1508 days)
343/606 — LNC (2020-01-02 to 2023-07-04) ... OK (881 days)
344/606 — LNT (2020-01-02 to 2026-01-01) ... OK (1508 days)
345/606 — LOW (2020-01-02 to 2026-01-01) ... OK (1508 days)
346/606 — LRCX (2020-01-02 to 2026-01-01) ... OK (1508 days)
347/606 — LULU (2024-01-02 to 2026-01

$MRO: possibly delisted; no timezone found

1 Failed download:
['MRO']: possibly delisted; no timezone found


FAILED
382/606 — MRSH (2020-01-02 to 2026-01-01) ... OK (1508 days)
383/606 — MS (2020-01-02 to 2026-01-01) ... OK (1508 days)
384/606 — MSCI (2020-01-02 to 2026-01-01) ... OK (1508 days)
385/606 — MSFT (2020-01-02 to 2026-01-01) ... OK (1508 days)
386/606 — MSI (2020-01-02 to 2026-01-01) ... OK (1508 days)
387/606 — MTB (2020-01-02 to 2026-01-01) ... OK (1508 days)
388/606 — MTCH (2021-10-01 to 2026-01-01) ... OK (1067 days)
389/606 — MTD (2020-01-02 to 2026-01-01) ... OK (1508 days)
390/606 — MU (2020-01-02 to 2026-01-01) ... OK (1508 days)
391/606 — MXIM (2020-01-02 to 2021-07-02) ... 

$MXIM: possibly delisted; no timezone found

1 Failed download:
['MXIM']: possibly delisted; no timezone found


FAILED
392/606 — MYL (2020-01-02 to 2020-10-02) ... 

$MYL: possibly delisted; no timezone found

1 Failed download:
['MYL']: possibly delisted; no timezone found


FAILED
393/606 — NBL (2020-01-02 to 2020-10-02) ... 

$NBL: possibly delisted; no timezone found

1 Failed download:
['NBL']: possibly delisted; no timezone found


FAILED
394/606 — NCLH (2020-01-02 to 2026-01-01) ... OK (1508 days)
395/606 — NDAQ (2020-01-02 to 2026-01-01) ... OK (1508 days)
396/606 — NDSN (2022-04-01 to 2026-01-01) ... OK (941 days)
397/606 — NEE (2020-01-02 to 2026-01-01) ... OK (1508 days)
398/606 — NEM (2020-01-02 to 2026-01-01) ... OK (1508 days)
399/606 — NFLX (2020-01-02 to 2026-01-01) ... OK (1508 days)
400/606 — NI (2020-01-02 to 2026-01-01) ... OK (1508 days)
401/606 — NKE (2020-01-02 to 2026-01-01) ... OK (1508 days)
402/606 — NLSN (2020-01-02 to 2022-10-04) ... 

$NLSN: possibly delisted; no timezone found

1 Failed download:
['NLSN']: possibly delisted; no timezone found


FAILED
403/606 — NOC (2020-01-02 to 2026-01-01) ... OK (1508 days)
404/606 — NOV (2020-01-02 to 2021-07-02) ... OK (378 days)
405/606 — NOW (2020-01-02 to 2026-01-01) ... OK (1508 days)
406/606 — NRG (2020-01-02 to 2026-01-01) ... OK (1508 days)
407/606 — NSC (2020-01-02 to 2026-01-01) ... OK (1508 days)
408/606 — NTAP (2020-01-02 to 2026-01-01) ... OK (1508 days)
409/606 — NTRS (2020-01-02 to 2026-01-01) ... OK (1508 days)
410/606 — NUE (2020-01-02 to 2026-01-01) ... OK (1508 days)
411/606 — NVDA (2020-01-02 to 2026-01-01) ... OK (1508 days)
412/606 — NVR (2020-01-02 to 2026-01-01) ... OK (1508 days)
413/606 — NWL (2020-01-02 to 2023-07-04) ... OK (881 days)
414/606 — NWS (2020-01-02 to 2026-01-01) ... OK (1508 days)
415/606 — NWSA (2020-01-02 to 2026-01-01) ... OK (1508 days)
416/606 — NXPI (2021-04-01 to 2026-01-01) ... OK (1194 days)
417/606 — O (2020-01-02 to 2026-01-01) ... OK (1508 days)
418/606 — ODFL (2020-01-02 to 2026-01-01) ... OK (1508 days)
419/606 — OGN (2021-07-01 to 20

$PARA: possibly delisted; no timezone found

1 Failed download:
['PARA']: possibly delisted; no timezone found


FAILED
429/606 — PAYC (2020-04-01 to 2026-01-01) ... OK (1446 days)
430/606 — PAYX (2020-01-02 to 2026-01-01) ... OK (1508 days)
431/606 — PBCT (2020-01-02 to 2022-04-02) ... 

$PBCT: possibly delisted; no timezone found

1 Failed download:
['PBCT']: possibly delisted; no timezone found


FAILED
432/606 — PCAR (2020-01-02 to 2026-01-01) ... OK (1508 days)
433/606 — PCG (2022-10-03 to 2026-01-01) ... OK (815 days)
434/606 — PEG (2020-01-02 to 2026-01-01) ... OK (1508 days)
435/606 — PENN (2021-04-01 to 2022-07-02) ... OK (316 days)
436/606 — PEP (2020-01-02 to 2026-01-01) ... OK (1508 days)
437/606 — PFE (2020-01-02 to 2026-01-01) ... OK (1508 days)
438/606 — PFG (2020-01-02 to 2026-01-01) ... OK (1508 days)
439/606 — PG (2020-01-02 to 2026-01-01) ... OK (1508 days)
440/606 — PGR (2020-01-02 to 2026-01-01) ... OK (1508 days)
441/606 — PH (2020-01-02 to 2026-01-01) ... OK (1508 days)
442/606 — PHM (2020-01-02 to 2026-01-01) ... OK (1508 days)
443/606 — PKG (2020-01-02 to 2026-01-01) ... OK (1508 days)
444/606 — PLD (2020-01-02 to 2026-01-01) ... OK (1508 days)
445/606 — PLTR (2024-10-01 to 2026-01-01) ... OK (314 days)
446/606 — PM (2020-01-02 to 2026-01-01) ... OK (1508 days)
447/606 — PNC (2020-01-02 to 2026-01-01) ... OK (1508 days)
448/606 — PNR (2020-01-02 to 2026-01

$PXD: possibly delisted; no timezone found

1 Failed download:
['PXD']: possibly delisted; no timezone found


FAILED
463/606 — PYPL (2020-01-02 to 2026-01-01) ... OK (1508 days)
464/606 — QCOM (2020-01-02 to 2026-01-01) ... OK (1508 days)
465/606 — QRVO (2020-01-02 to 2024-10-02) ... OK (1195 days)
466/606 — RCL (2020-01-02 to 2026-01-01) ... OK (1508 days)
467/606 — REG (2020-01-02 to 2026-01-01) ... OK (1508 days)
468/606 — REGN (2020-01-02 to 2026-01-01) ... OK (1508 days)
469/606 — RF (2020-01-02 to 2026-01-01) ... OK (1508 days)
470/606 — RHI (2020-01-02 to 2024-04-02) ... OK (1068 days)
471/606 — RJF (2020-01-02 to 2026-01-01) ... OK (1508 days)
472/606 — RL (2020-01-02 to 2026-01-01) ... OK (1508 days)
473/606 — RMD (2020-01-02 to 2026-01-01) ... OK (1508 days)
474/606 — ROK (2020-01-02 to 2026-01-01) ... OK (1508 days)
475/606 — ROL (2020-01-02 to 2026-01-01) ... OK (1508 days)
476/606 — ROP (2020-01-02 to 2026-01-01) ... OK (1508 days)
477/606 — ROST (2020-01-02 to 2026-01-01) ... OK (1508 days)
478/606 — RSG (2020-01-02 to 2026-01-01) ... OK (1508 days)
479/606 — RTN (2020-01-02 to 2

$RTN: possibly delisted; no timezone found

1 Failed download:
['RTN']: possibly delisted; no timezone found


FAILED
480/606 — RTX (2020-01-02 to 2026-01-01) ... OK (1508 days)
481/606 — RVTY (2020-01-02 to 2026-01-01) ... OK (1508 days)
482/606 — SBAC (2020-01-02 to 2026-01-01) ... OK (1508 days)
483/606 — SBNY (2022-01-03 to 2023-01-04) ... 

$SBNY: possibly delisted; no price data found  (1d 2022-01-03 00:00:00 -> 2023-01-04 00:00:00) (Yahoo error = "Data doesn't exist for startDate = 1641186000, endDate = 1672808400")

1 Failed download:
['SBNY']: possibly delisted; no price data found  (1d 2022-01-03 00:00:00 -> 2023-01-04 00:00:00) (Yahoo error = "Data doesn't exist for startDate = 1641186000, endDate = 1672808400")


FAILED
484/606 — SBUX (2020-01-02 to 2026-01-01) ... OK (1508 days)
485/606 — SCHW (2020-01-02 to 2026-01-01) ... OK (1508 days)
486/606 — SEDG (2022-01-03 to 2023-10-03) ... OK (439 days)
487/606 — SEE (2020-01-02 to 2023-10-03) ... OK (944 days)
488/606 — SHW (2020-01-02 to 2026-01-01) ... OK (1508 days)
489/606 — SIVBQ (2020-01-02 to 2023-01-04) ... 

$SIVBQ: possibly delisted; no timezone found

1 Failed download:
['SIVBQ']: possibly delisted; no timezone found


FAILED
490/606 — SJM (2020-01-02 to 2026-01-01) ... OK (1508 days)
491/606 — SLB (2020-01-02 to 2026-01-01) ... OK (1508 days)
492/606 — SLG (2020-01-02 to 2021-01-05) ... OK (254 days)
493/606 — SLVM (2021-10-01 to 2021-10-02) ... OK (1 days)
494/606 — SMCI (2024-04-01 to 2026-01-01) ... OK (441 days)
495/606 — SNA (2020-01-02 to 2026-01-01) ... OK (1508 days)
496/606 — SNPS (2020-01-02 to 2026-01-01) ... OK (1508 days)
497/606 — SO (2020-01-02 to 2026-01-01) ... OK (1508 days)
498/606 — SOLV (2024-04-01 to 2026-01-01) ... OK (441 days)
499/606 — SPG (2020-01-02 to 2026-01-01) ... OK (1508 days)
500/606 — SPGI (2020-01-02 to 2026-01-01) ... OK (1508 days)
501/606 — SRE (2020-01-02 to 2026-01-01) ... OK (1508 days)
502/606 — STE (2020-01-02 to 2026-01-01) ... OK (1508 days)
503/606 — STLD (2023-01-03 to 2026-01-01) ... OK (752 days)
504/606 — STT (2020-01-02 to 2026-01-01) ... OK (1508 days)
505/606 — STX (2020-01-02 to 2026-01-01) ... OK (1508 days)
506/606 — STZ (2020-01-02 to 2026-0

$TWTR: possibly delisted; no timezone found

1 Failed download:
['TWTR']: possibly delisted; no timezone found


FAILED
540/606 — TXN (2020-01-02 to 2026-01-01) ... OK (1508 days)
541/606 — TXT (2020-01-02 to 2026-01-01) ... OK (1508 days)
542/606 — TYL (2020-07-01 to 2026-01-01) ... OK (1383 days)
543/606 — UA (2020-01-02 to 2022-04-02) ... OK (568 days)
544/606 — UAA (2020-01-02 to 2022-04-02) ... OK (568 days)
545/606 — UAL (2020-01-02 to 2026-01-01) ... OK (1508 days)
546/606 — UBER (2024-01-02 to 2026-01-01) ... OK (502 days)
547/606 — UDR (2020-01-02 to 2026-01-01) ... OK (1508 days)
548/606 — UHS (2020-01-02 to 2026-01-01) ... OK (1508 days)
549/606 — ULTA (2020-01-02 to 2026-01-01) ... OK (1508 days)
550/606 — UNH (2020-01-02 to 2026-01-01) ... OK (1508 days)
551/606 — UNM (2020-01-02 to 2021-07-02) ... OK (378 days)
552/606 — UNP (2020-01-02 to 2026-01-01) ... OK (1508 days)
553/606 — UPS (2020-01-02 to 2026-01-01) ... OK (1508 days)
554/606 — URI (2020-01-02 to 2026-01-01) ... OK (1508 days)
555/606 — USB (2020-01-02 to 2026-01-01) ... OK (1508 days)
556/606 — V (2020-01-02 to 2026-01-0

$VAR: possibly delisted; no timezone found

1 Failed download:
['VAR']: possibly delisted; no timezone found


FAILED
558/606 — VFC (2020-01-02 to 2024-04-02) ... OK (1068 days)
559/606 — VICI (2022-07-01 to 2026-01-01) ... OK (879 days)
560/606 — VLO (2020-01-02 to 2026-01-01) ... OK (1508 days)
561/606 — VLTO (2023-10-02 to 2026-01-01) ... OK (563 days)
562/606 — VMC (2020-01-02 to 2026-01-01) ... OK (1508 days)
563/606 — VNO (2020-01-02 to 2023-01-04) ... OK (757 days)
564/606 — VNT (2021-01-04 to 2021-01-05) ... OK (1 days)
565/606 — VRSK (2020-01-02 to 2026-01-01) ... OK (1508 days)
566/606 — VRSN (2020-01-02 to 2026-01-01) ... OK (1508 days)
567/606 — VRTX (2020-01-02 to 2026-01-01) ... OK (1508 days)
568/606 — VST (2024-07-01 to 2026-01-01) ... OK (378 days)
569/606 — VTR (2020-01-02 to 2026-01-01) ... OK (1508 days)
570/606 — VTRS (2021-01-04 to 2026-01-01) ... OK (1255 days)
571/606 — VZ (2020-01-02 to 2026-01-01) ... OK (1508 days)
572/606 — WAB (2020-01-02 to 2026-01-01) ... OK (1508 days)
573/606 — WAT (2020-01-02 to 2026-01-01) ... OK (1508 days)
574/606 — WBA (2020-01-02 to 2025-0

$WBA: possibly delisted; no timezone found

1 Failed download:
['WBA']: possibly delisted; no timezone found


FAILED
575/606 — WBD (2022-07-01 to 2026-01-01) ... OK (879 days)
576/606 — WCG (2020-01-02 to 2020-01-03) ... 

$WCG: possibly delisted; no timezone found

1 Failed download:
['WCG']: possibly delisted; no timezone found


FAILED
577/606 — WDAY (2025-01-02 to 2026-01-01) ... OK (250 days)
578/606 — WDC (2020-01-02 to 2026-01-01) ... OK (1508 days)
579/606 — WEC (2020-01-02 to 2026-01-01) ... OK (1508 days)
580/606 — WELL (2020-01-02 to 2026-01-01) ... OK (1508 days)
581/606 — WFC (2020-01-02 to 2026-01-01) ... OK (1508 days)
582/606 — WHR (2020-01-02 to 2024-01-03) ... OK (1007 days)
583/606 — WM (2020-01-02 to 2026-01-01) ... OK (1508 days)
584/606 — WMB (2020-01-02 to 2026-01-01) ... OK (1508 days)
585/606 — WMT (2020-01-02 to 2026-01-01) ... OK (1508 days)
586/606 — WRB (2020-01-02 to 2026-01-01) ... OK (1508 days)
587/606 — WRK (2020-01-02 to 2024-07-02) ... 

$WRK: possibly delisted; no timezone found

1 Failed download:
['WRK']: possibly delisted; no timezone found


FAILED
588/606 — WSM (2025-04-01 to 2026-01-01) ... OK (190 days)
589/606 — WST (2020-07-01 to 2026-01-01) ... OK (1383 days)
590/606 — WTW (2020-01-02 to 2026-01-01) ... OK (1508 days)
591/606 — WU (2020-01-02 to 2021-10-02) ... OK (442 days)
592/606 — WY (2020-01-02 to 2026-01-01) ... OK (1508 days)
593/606 — WYNN (2020-01-02 to 2026-01-01) ... OK (1508 days)
594/606 — XEC (2020-01-02 to 2020-01-03) ... 

$XEC: possibly delisted; no timezone found

1 Failed download:
['XEC']: possibly delisted; no timezone found


FAILED
595/606 — XEL (2020-01-02 to 2026-01-01) ... OK (1508 days)
596/606 — XLNX (2020-01-02 to 2022-01-04) ... 

$XLNX: possibly delisted; no timezone found

1 Failed download:
['XLNX']: possibly delisted; no timezone found


FAILED
597/606 — XOM (2020-01-02 to 2026-01-01) ... OK (1508 days)
598/606 — XRAY (2020-01-02 to 2024-04-02) ... OK (1068 days)
599/606 — XRX (2020-01-02 to 2021-01-05) ... OK (254 days)
600/606 — XYL (2020-01-02 to 2026-01-01) ... OK (1508 days)
601/606 — XYZ (2025-10-01 to 2026-01-01) ... OK (64 days)
602/606 — YUM (2020-01-02 to 2026-01-01) ... OK (1508 days)
603/606 — ZBH (2020-01-02 to 2026-01-01) ... OK (1508 days)
604/606 — ZBRA (2020-01-02 to 2026-01-01) ... OK (1508 days)
605/606 — ZION (2020-01-02 to 2024-01-03) ... OK (1007 days)
606/606 — ZTS (2020-01-02 to 2026-01-01) ... OK (1508 days)

Success: 560
Failed: 46
Failed tickers: ['ABMD', 'AGN', 'ALXN', 'ANSS', 'ARNC', 'ATVI', 'CERN', 'CMA', 'CTLT', 'CTXS', 'CXO', 'DAY', 'DFS', 'DISCA', 'DISCK', 'DISH', 'DRE', 'ETFC', 'FLIR', 'HBI', 'HES', 'HFC', 'IPG', 'JNPR', 'JWN', 'K', 'KLG', 'KSU', 'MRO', 'MXIM', 'MYL', 'NBL', 'NLSN', 'PARA', 'PBCT', 'PXD', 'RTN', 'SBNY', 'SIVBQ', 'TWTR', 'VAR', 'WBA', 'WCG', 'WRK', 'XEC', 'XLNX']


In [6]:
# Verify the saved file
prices = pd.read_csv(r"..\..\Data\Main\sp500_prices_2020_2025.csv")
print(f"Shape: {prices.shape}")
print(f"Tickers: {prices['ticker'].nunique()}")
print(f"Date range: {prices['date'].min()} to {prices['date'].max()}")
print(f"Columns: {list(prices.columns)}")

success_df.to_csv(r"..\..\Data\Main\price_download_log.csv", index=False)
failed_df = pd.DataFrame({'ticker': failed})
failed_df.to_csv(r"..\..\Data\Main\price_download_failed.csv", index=False)

print(f"\nSaved:")
print(f"  sp500_prices_2020_2025.csv — {len(prices)} rows, {prices['ticker'].nunique()} tickers")
print(f"  price_download_log.csv — {len(success_df)} successful tickers")
print(f"  price_download_failed.csv — {len(failed_df)} failed tickers")

Shape: (722756, 7)
Tickers: 560
Date range: 2020-01-02 to 2025-12-31
Columns: ['date', 'Close', 'High', 'Low', 'Open', 'Volume', 'ticker']

Saved:
  sp500_prices_2020_2025.csv — 722756 rows, 560 tickers
  price_download_log.csv — 560 successful tickers
  price_download_failed.csv — 46 failed tickers


# Checking Failed Ticker Quarter Duration

In [7]:
# Checking how long these tickers were in the index
failed_panel = panel[panel['yf_ticker'].isin(failed)]
print(f"Failed tickers: {len(failed)}")
print(f"\nMembership duration of failed tickers:")
print(failed_panel[['yf_ticker', 'name', 'first_seen', 'last_seen', 'quarters_present']].to_string(index=False))

Failed tickers: 46

Membership duration of failed tickers:
yf_ticker                           name first_seen  last_seen  quarters_present
     ABMD                    ABIOMED Inc 2020-01-02 2022-10-03                12
      AGN              Allergan UnLtd Co 2020-01-02 2020-04-01                 2
     ALXN Alexion Pharmaceuticals Inc/MA 2020-01-02 2021-07-01                 7
     ANSS                      ANSYS Inc 2020-01-02 2025-07-01                23
     ARNC                   Arconic Corp 2020-04-01 2020-04-01                 1
     ATVI        Activision Blizzard Inc 2020-01-02 2023-10-02                16
     CERN                    Cerner Corp 2020-01-02 2022-04-01                10
      CMA                   Comerica Inc 2020-01-02 2024-04-01                18
     CTLT                   Catalent Inc 2020-10-01 2024-10-01                17
     CTXS             Citrix Systems Inc 2020-01-02 2022-07-01                11
      CXO           Concho Resources Inc 2020-01-0

# Categorise Failed Tickers by Membership Duration

In [8]:
failed_panel = panel[panel['yf_ticker'].isin(failed)].copy()

short = failed_panel[failed_panel['quarters_present'] <= 4]
medium = failed_panel[(failed_panel['quarters_present'] > 4) & (failed_panel['quarters_present'] <= 12)]
long_term = failed_panel[failed_panel['quarters_present'] > 12]

print(f"Short membership (1-4 quarters): {len(short)} tickers")
print(short[['yf_ticker', 'name', 'quarters_present']].to_string(index=False))

print(f"\nMedium membership (5-12 quarters): {len(medium)} tickers")
print(medium[['yf_ticker', 'name', 'quarters_present']].to_string(index=False))

print(f"\nLong membership (13+ quarters): {len(long_term)} tickers")
print(long_term[['yf_ticker', 'name', 'quarters_present']].to_string(index=False))

Short membership (1-4 quarters): 10 tickers
yf_ticker                        name  quarters_present
      AGN           Allergan UnLtd Co                 2
     ARNC                Arconic Corp                 1
     ETFC      E*TRADE Financial Corp                 4
      JWN               Nordstrom Inc                 2
      KLG               WK Kellogg Co                 1
      MYL                    Mylan NV                 4
      NBL            Noble Energy Inc                 4
      RTN                 Raytheon Co                 2
      WCG   WellCare Health Plans Inc                 1
      XEC Coterra Energy Operating Co                 1

Medium membership (5-12 quarters): 19 tickers
yf_ticker                           name  quarters_present
     ABMD                    ABIOMED Inc                12
     ALXN Alexion Pharmaceuticals Inc/MA                 7
     CERN                    Cerner Corp                10
     CTXS             Citrix Systems Inc                1

# Quantify Data Loss from Excluded Tickers

In [9]:
success_panel = panel[panel['yf_ticker'].isin([s['ticker'] for s in success])]

failed_panel['est_trading_days'] = failed_panel['quarters_present'] * 63
total_possible = panel['quarters_present'].sum() * 63
total_lost = failed_panel['est_trading_days'].sum()

print(f"Total tickers in panel: {len(panel)}")
print(f"Successfully downloaded: {len(success)}")
print(f"Failed: {len(failed)} ({len(failed)/len(panel)*100:.1f}%)")
print(f"\nEstimated trading days lost: {total_lost:,} out of {total_possible:,} ({total_lost/total_possible*100:.1f}%)")

Total tickers in panel: 606
Successfully downloaded: 560
Failed: 46 (7.6%)

Estimated trading days lost: 33,012 out of 761,418 (4.3%)


# Investigate Failed Tickers via Yahoo Finance Search API

In [10]:
import requests
import time

print("Investigating each failed ticker via Yahoo Finance search:\n")
results = []

for _, row in failed_panel.iterrows():
    t = row['yf_ticker']
    name = row['name']
    
    try:
        url = f"https://query2.finance.yahoo.com/v1/finance/search?q={name}&quotesCount=3"
        headers = {'User-Agent': 'Mozilla/5.0'}
        response = requests.get(url, headers=headers)
        quotes = response.json().get('quotes', [])
        
        if quotes:
            top = quotes[0]
            symbol = top['symbol']
            match_name = top.get('longname', top.get('shortname', 'N/A'))
            print(f"{t:>6} ({name})")
            print(f"       → {symbol} ({match_name})")
        else:
            print(f"{t:>6} ({name})")
            print(f"       → No results — likely fully delisted or taken private")
        
        results.append({
            'ticker': t,
            'name': name,
            'quarters_present': row['quarters_present'],
            'last_seen': row['last_seen'],
            'yf_search_result': quotes[0]['symbol'] if quotes else 'NONE',
            'yf_search_name': quotes[0].get('longname', 'N/A') if quotes else 'N/A'
        })
    except Exception as e:
        print(f"{t:>6} ({name}) → Error: {e}")
        results.append({
            'ticker': t, 'name': name,
            'quarters_present': row['quarters_present'],
            'last_seen': row['last_seen'],
            'yf_search_result': 'ERROR', 'yf_search_name': str(e)
        })
    
    time.sleep(0.5)

results_df = pd.DataFrame(results)
results_df.to_csv(r"..\..\Data\Main\failed_tickers_investigation.csv", index=False)
print(f"\nSaved to failed_tickers_investigation.csv")

Investigating each failed ticker via Yahoo Finance search:

  ABMD (ABIOMED Inc)
       → No results — likely fully delisted or taken private
   AGN (Allergan UnLtd Co)
       → No results — likely fully delisted or taken private
  ALXN (Alexion Pharmaceuticals Inc/MA)
       → No results — likely fully delisted or taken private
  ANSS (ANSYS Inc)
       → No results — likely fully delisted or taken private
  ARNC (Arconic Corp)
       → No results — likely fully delisted or taken private
  ATVI (Activision Blizzard Inc)
       → No results — likely fully delisted or taken private
  CERN (Cerner Corp)
       → No results — likely fully delisted or taken private
   CMA (Comerica Inc)
       → No results — likely fully delisted or taken private
  CTLT (Catalent Inc)
       → No results — likely fully delisted or taken private
  CTXS (Citrix Systems Inc)
       → No results — likely fully delisted or taken private
   CXO (Concho Resources Inc)
       → No results — likely fully delisted o

# Review Investigation Results

In [11]:
no_results = results_df[results_df['yf_search_result'] == 'NONE']
has_results = results_df[results_df['yf_search_result'] != 'NONE']

print(f"Tickers fully removed from Yahoo Finance: {len(no_results)} out of {len(results_df)}")
print(f"\nTickers with search results ({len(has_results)}):")
for _, row in has_results.iterrows():
    print(f"  {row['ticker']} → {row['yf_search_result']} ({row['yf_search_name']})")

Tickers fully removed from Yahoo Finance: 42 out of 46

Tickers with search results (4):
  DISCA → WBD (Warner Bros. Discovery, Inc.)
  DISCK → WBD (Warner Bros. Discovery, Inc.)
  SBNY → SBNYL (Signature Bank)
  WRK → WEST (Westrock Coffee Company)


# Investigation Findings

**Yahoo Finance search results:**
- DISCA/DISCK → WBD (Warner Bros Discovery): Discovery Inc share classes that merged into WBD, which is already in our panel. No data loss.
- SBNY → SBNYL: Only a residual security remains. Common stock data was removed after the bank collapsed in March 2023.
- WRK → WEST: False match. Westrock Coffee is a different company from WestRock Co (packaging), which merged with Smurfit Kappa in July 2024.

**All remaining 42 tickers returned no results — fully removed from Yahoo Finance.**
46 tickers returned no data from Yahoo Finance. All 46 display "possibly delisted" errors from yfinance, indicating their securities have been removed from Yahoo Finance's database.Three (DAY, IPG, K) were verified as recent acquisitions via Bloomberg Terminal DES command:
- DAY (Dayforce) — acquired by Thoma Bravo, completed Feb 2026
- IPG (Interpublic) — acquired by Omnicom Group, completed Nov 2025
- K (Kellanova) — acquired by Mars Inc, completed Dec 2025

The remaining 39 are consistent with companies acquired, taken private, or otherwise delisted during the study period. Total estimated data loss is 4.3% of trading days — insufficient to introduce systematic bias.

All 46 excluded from analysis. Remaining: 560 tickers with price data.